### PR TITLE
fix: allow for bare variable substitution

### PR DESCRIPTION
**Pull request by kellyjosephprice**
## 🧰 What's being changed?

ReadMe user variables are written like `<<var>>`. Depending on the
language parser used by CodeMirror, that could get parsed into
multiple tokens, ie [`<<`, `var`, `>>`]. So, we do a first pass to
remove the variables. And reinsert them after CodeMirror has done its
parsing.

This PR also fixes inserting multiple variables per line.

## 🧪 Testing

The following would not get interpolated:

```js
const foo = <<name>>;
```

Multiple variables (previously, only the second would get interpolated):
```js
const foo = { bar: <<name>>, baz: <<name>> };
```

[[bare variable ticket]](https://app.asana.com/0/1199089258199423/1198973877710478/f)
[[multiple variable ticket]](https://app.asana.com/0/1178132843162889/1179842282864914/f)

### CODE REVIEW

File: __tests__/codeMirror.test.js [hunk 1]
```diff
@@ -62,6 +62,27 @@ test('should keep enclosing characters around the variable', () => {
   expect(mount(syntaxHighlighter("'<<apiKey>>'", 'json', { tokenizeVariables: true })).text()).toBe("'APIKEY'");
 });
 
+test('should tokenize variables outside of quotes', () => {
+  expect(mount(syntaxHighlighter('<<apiKey>>', 'json', { tokenizeVariables: true })).text()).toBe('APIKEY');
+});
+
+test('should tokenize variables outside of quotes over multple lines', () => {
```
Comment (domharrington) [inline __tests__/codeMirror.test.js:69]:
```suggestion
test('should tokenize variables outside of quotes over multiple lines', () => {
```

File: __tests__/__snapshots__/codeMirror.test.js.snap [hunk 2]
```diff
@@ -346,3 +346,12 @@ exports[`Supported languages cURL should syntax highlight an example 1`] = `
   <span class=\\"cm-attribute\\">--header</span> <span class=\\"cm-string\\">&#x27;x-readme-version: v3.0&#x27;</span>
 </div>"
 `;
+
+exports[`should tokenize variables outside of quotes over multple lines 1`] = `
```
Comment (domharrington) [inline __tests__/__snapshots__/codeMirror.test.js.snap:350]:
```suggestion
exports[`should tokenize variables outside of quotes over multiple lines 1`] = `
```

**Review by domharrington — APPROVED** [review]
Well that's a really interesting problem to have to solve! Code looks good. As long as it's been tested against readme with all of these different states then i'm happy.

**Comment by kellyjosephprice** [comment]
before | after
------ | -----
Screenshot_2020-12-03 Testing variables(1) | Screenshot_2020-12-03 Testing variables

**Pull request merged**